pyproject.toml

```diff
@@ -62,3 +62,90 @@ strict_equality = true
 [[tool.mypy.overrides]]
 module = ["django.*"]
 ignore_missing_imports = true
+
+[tool.ruff]
+line-length = 120
+target-version = "py38"
+
+[tool.ruff.lint]
+select = [
+  "AIR",    # Airflow
+  "ASYNC",  # flake8-async
+  "BLE",    # flake8-blind-except
+  "C90",    # McCabe cyclomatic complexity
+  "DJ",     # flake8-django
+  "DTZ",    # flake8-datetimez
+  "E",      # pycodestyle errors
+  "F",      # Pyflakes
+  "FIX",    # flake8-fixme
+  "FLY",    # flynt
+  "G",      # flake8-logging-format
+  "ICN",    # flake8-import-conventions
+  "INP",    # flake8-no-pep420
+  "INT",    # flake8-gettext
+  "NPY",    # NumPy-specific rules
+  "PD",     # pandas-vet
+  "PIE",    # flake8-pie
+  "PL",     # Pylint
+  "PYI",    # flake8-pyi
+  "RSE",    # flake8-raise
+  "SLOT",   # flake8-slots
+  "T10",    # flake8-debugger
+  "T20",    # flake8-print
+  "TD",     # flake8-todos
+  "TID",    # flake8-tidy-imports
+  "UP",     # pyupgrade
+  "W",      # pycodestyle warnings
+  "YTT",    # flake8-2020
+  # "A",    # flake8-builtins
+  # "ANN",  # flake8-annotations
+  # "ARG",  # flake8-unused-arguments
+  # "B",    # flake8-bugbear
+  # "C4",   # flake8-comprehensions
+  # "COM",  # flake8-commas
+  # "CPY",    # flake8-copyright
+  # "D",    # pydocstyle
+  # "EM",   # flake8-errmsg
+  # "ERA",  # eradicate
+  # "EXE",  # flake8-executable
+  # "FA",   # flake8-future-annotations
+  # "FBT",  # flake8-boolean-trap
+  # "I",    # isort
+  # "ISC",  # flake8-implicit-str-concat
+  # "N",    # pep8-naming
+  # "PERF", # Perflint
+  # "PGH",  # pygrep-hooks
+  # "PT",   # flake8-pytest-style
+  # "PTH",  # flake8-use-pathlib
+  # "Q",    # flake8-quotes
+  # "RET",  # flake8-return
+  # "RUF",  # Ruff-specific rules
+  # "S",    # flake8-bandit
+  # "SIM",  # flake8-simplify
+  # "SLF",  # flake8-self
+  # "TCH",  # flake8-type-checking
+  # "TRY",  # tryceratops
+]
+# Files not checked:
+#  - migrations: most of these are autogenerated and don't need a check
+#  - docs: contains autogenerated code that doesn't need a check
+exclude = [
+    "*/migrations/*",
+    "docs",
+]
+ignore = ["F401"]
+
+[tool.ruff.lint.mccabe]
+max-complexity = 23
+
+[tool.ruff.lint.per-file-ignores]
+"docs/conf.py" = ["INP001"]
+"test_app/models.py" = ["DJ008"]  # FIXME
+"waffle/models.py" = ["DJ012", "PYI019"]  # FIXME
+
+[tool.ruff.lint.pylint]
+allow-magic-value-types = ["float", "int", "str"]
+max-args = 6  # default is 5
```

Comment:
praise: nice to have the defaults mentioned here as a possible future target for refactoring